test/celeritas/optical/ValidationUtils.cc

```diff
@@ -0,0 +1,111 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/ValidationUtils.cc
+//---------------------------------------------------------------------------//
+#include "ValidationUtils.hh"
+
+namespace celeritas
+{
+namespace optical
+{
+namespace test
+{
+using namespace ::celeritas::test;
+//---------------------------------------------------------------------------//
+/*!
+ */
+void check_physics_vector(ImportPhysicsVector const& expected,
+                          ImportPhysicsVector const& actual)
+{
+    EXPECT_EQ(expected.vector_type, actual.vector_type);
+    EXPECT_VEC_EQ(expected.x, actual.x);
+    EXPECT_VEC_EQ(expected.y, actual.y);
```

Comment:
The problem with this is that errors now look like they're coming from this file instead of the source where the test occurs. Another issue is that you're using exact equality for floating point values. If you take a look at `test/TestMacros.hh` you can see how to correctly do this with googletest (and the implementation is in `test/testdetail/TestMacrosImpl.hh`). To make life easier, I think you should just overload the `IsVecSoftEquiv` function,  declaring it inside ValidationUtils.hh
```c++
::testing::AssertionResult IsVecSoftEquiv(char const* expected_expr,
                                          char const* actual_expr,
                                          ImportPhysicsVector const& expected,
                                          ImportPhysicsVector const& actual)
```

Reply:
I was using exact equality here just to check that imported physics vectors get forwarded to adapters / models correctly. Since they're only copies and references of doubles to doubles, I believe exact equality should be the desired behavior.